middleware.ts

```diff
@@ -0,0 +1 @@
+export { locales as middleware } from "./locales";
```

Comment:
Are locales here used by Next.js automagically?

Reply:
Yes, https://nextra.site/docs/guide/i18n

due to the locales module Nextra provided is not that flexible, I optimized it to more be suit for docs